Enhance documentation and functionality for trace management

Dwij1704 opened the pull request:
- Updated `mint.json` to include new usage paths for trace decorators and manual trace control.
- Revised `introduction.mdx` to reflect changes in session tracking terminology, replacing "sessions" with "traces" for clarity.
- Expanded `quickstart.mdx` to introduce the `@trace` decorator and manual trace management examples.
- Added new documentation files for `manual-trace-control.mdx` and `trace-decorator.mdx`, detailing advanced trace management techniques and usage patterns.
- Updated `sdk-reference.mdx` to include new parameters for trace management and improved examples for initializing the AgentOps SDK.
- Enhanced `tracking-agents.mdx` to demonstrate multi-agent workflows and coordination using the `@trace` decorator.

These changes improve the clarity and usability of the AgentOps SDK, particularly around trace management and agent tracking.

## 📥 Pull Request

**📘 Description**
_Briefly describe the changes you've made._

**🧪 Testing**
_Describe the tests you performed to validate your changes._

Dwij1704 committed:
Enhance documentation and functionality for trace management

- Updated `mint.json` to include new usage paths for trace decorators and manual trace control.
- Revised `introduction.mdx` to reflect changes in session tracking terminology, replacing "sessions" with "traces" for clarity.
- Expanded `quickstart.mdx` to introduce the `@trace` decorator and manual trace management examples.
- Added new documentation files for `manual-trace-control.mdx` and `trace-decorator.mdx`, detailing advanced trace management techniques and usage patterns.
- Updated `sdk-reference.mdx` to include new parameters for trace management and improved examples for initializing the AgentOps SDK.
- Enhanced `tracking-agents.mdx` to demonstrate multi-agent workflows and coordination using the `@trace` decorator.

These changes improve the clarity and usability of the AgentOps SDK, particularly around trace management and agent tracking.

areibman commented:
You the man Dwij! Gonna review this later

areibman commented on docs/v2/usage/manual-trace-control.mdx:
```diff
@@ -0,0 +1,288 @@
+agentops.end_trace(end_state="Emergency_Shutdown")
+
+# Or with a different state
+agentops.end_trace(end_state="Timeout")
```
These states (i.e. Timeout, Emergency Shutdown) don't actually exist in the source code. They're just arbitrary strings. We shouldn't make users believe they are reserved keywords

areibman commented on docs/v2/usage/manual-trace-control.mdx:
```diff
@@ -0,0 +1,288 @@
+agentops.end_trace(trace, "Unauthorized")
+
+# Custom states
+agentops.end_trace(trace, "Custom_Business_State")
```
While a nice idea, our SDK doesn't actually support these! LLM slop :)

areibman commented on docs/v2/quickstart.mdx:
```diff
@@ -84,21 +84,43 @@ import EnvTooltip from '/snippets/add-env-tooltip.mdx'
+  # Or manage traces manually
+  import agentops
+  
+  trace = agentops.start_trace("custom-trace")
+  # Your code here
+  agentops.end_trace(trace, "Success")
```
Don't we need to run agentops.init?

areibman commented on docs/v2/usage/manual-trace-control.mdx:
```diff
@@ -0,0 +1,288 @@
+description: "Advanced trace management with start_trace and end_trace methods"
+---
+
+AgentOps 0.4.13 introduces powerful manual trace control capabilities, allowing you to precisely manage trace lifecycles and states. This is particularly useful for complex workflows, batch processing, and scenarios requiring fine-grained control over trace boundaries.
```
This version will change, so remove this paragraph.

areibman commented on docs/v2/usage/manual-trace-control.mdx:
```diff
@@ -0,0 +1,288 @@
+    batch_traces.append((trace, item))
+
+# Process each item and end traces individually
+for trace, item in batch_traces:
```
Does this actually work?

Dwij1704 replied:
Yes

areibman commented on docs/v2/usage/recording-operations.mdx:
````diff
@@ -86,24 +95,59 @@ def research_workflow(topic):
+    agent = ResearchAgent()
+    results = agent.search(f"customer {customer_id}")
+    return results
+```
````
Trace context management via decorators should probably be more widespread in our docs.

agentops.start_trace +  agentops.end_trace is potentially very dangerous when you have multiple traces in a program. In fact, production apps will generally be Fast API, django, flask, etc. which will necessarily be multi-trace.

So let's adapt the docs to generally adopt the @trace pattern

areibman commented on docs/v2/usage/recording-operations.mdx:
```diff
@@ -86,24 +95,59 @@ def research_workflow(topic):
+
+# End the trace with a specific state
+agentops.end_trace(trace, "Success")
+
```
See comment above, let's use decorators liberally

areibman commented on docs/v2/usage/recording-operations.mdx:
```diff
@@ -86,24 +95,59 @@ def research_workflow(topic):
+agentops.end_trace(trace, "Success")
+
+# Or end all active traces
+agentops.end_trace(end_state="Emergency_Shutdown")
```
See comment above-- "Emergency Shutdown" isn't a real state afaik

areibman commented on docs/v2/usage/trace-decorator.mdx:
```diff
@@ -0,0 +1,453 @@
+# Initialize AgentOps
+agentops.init("your-api-key", auto_start_session=False)
+
+@trace
```
I think the very first example should include names + tags. This helps us shorten the file and demonstrates the capabilities.

We can mention they are optional args.

areibman commented on docs/v2/usage/trace-decorator.mdx:
```diff
@@ -0,0 +1,453 @@
+    @operation
+    def generate_report(self, analysis):
+        return f"Report: {analysis}"
+
```
This agent should absolutely be making LLM calls. Let's add in a function that does an openai chat completion (sort of the whole point)

areibman commented on docs/v2/usage/trace-decorator.mdx:
````diff
@@ -0,0 +1,453 @@
+result = process_order(order)
+```
+
+### Machine Learning Pipeline
````
Doesn't seem like a great example for the agents we deal with. Let's scrap it.

areibman commented on docs/v2/usage/tracking-agents.mdx:
```diff
@@ -88,4 +204,34 @@ All operations are automatically associated with the agent that originated them.
+@trace(name="my-workflow")
+def my_workflow():
+    # workflow code
+    pass
```
Let's swap (new should be first)

areibman reviewed (changes requested):
Dwij, as always, you produce great docs. Left some comments in here for suggested improvements

Dwij1704 committed:
Enhance documentation for trace management and agent tracking

- Updated `introduction.mdx` to clarify the use of the `@trace` decorator for custom trace creation.
- Revised `quickstart.mdx` to include examples of initializing AgentOps with manual trace control.
- Improved `examples.mdx` by updating image links to use raw GitHub URLs for better accessibility.
- Enhanced `manual-trace-control.mdx` to provide detailed examples of managing traces manually.
- Updated `recording-operations.mdx` to emphasize the use of the `@trace` decorator for grouping operations.
- Refined `sdk-reference.mdx` to clarify parameters for ending traces and the use of decorators.
- Improved `tracking-agents.mdx` to demonstrate structured tracking with the `@agent` decorator in multi-agent systems.

These changes improve the clarity and usability of the AgentOps SDK, particularly around trace management and agent tracking.

Dwij1704 committed:
Add Google Generative AI instrumentation module

- Introduced a new module for Google Generative AI (Gemini) API instrumentation, including `GoogleGenAIInstrumentor` for capturing telemetry data.
- Added attribute extraction functions for model and chat interactions, enhancing observability of API calls.
- Implemented wrappers for synchronous and asynchronous streaming methods to track performance metrics.
- Updated documentation to include usage examples and details on supported features for the new instrumentation.
- Refactored existing code to accommodate the new module structure and improve clarity.

These changes enhance the AgentOps SDK by providing comprehensive support for monitoring Google Generative AI interactions.

Dwij1704 committed:
Update mint.json to reorder integration paths, moving 'anthropic' to the correct position in the integrations list. This change improves the organization and accessibility of integration documentation.

Dwij1704 commented:
Renamed google-generativeai to google-genai in this PR @bboynton97
